### PR TITLE
Feature added: API Register

### PR DESCRIPTION
**Pull request by codingwill**
I have created several files and features as follows:

Files created:
- api/Http/Controllers/**CustomerController.php**
- api/Models/**Customers.php**
- api/Http/Middleware/**JwtMiddleware.php**
- config/**jwt.php**
- database migrations for **customers** and **verification_tokens** table

Features added:
1. **JWTAuth:** Developers can now use JWT based authentication in this project.
2. **Register:** Customers can now register via endpoint **/api/v1/user/register**.
Corresponding function/method that handling this feature: **CustomerController::register()**

### CODE REVIEW

File: app/Http/Controllers/CustomerController.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{
+    
+    // User Registration
+    public function register(Request $request)
+    {
+        // validating email and password as per rules
+        $validator = Validator::make($request->all(), [
+            'name' => 'required|string|max:64',
+            'email' => 'required|string|email|max:64|unique:customers',
+            'password' => 'required|string|min:8|confirmed|regex:/^(?=.*[0-9])(?=.*[a-zA-Z])([a-zA-Z0-9]+)$/',
```
Comment (cakazies) [inline app/Http/Controllers/CustomerController.php:22]: ini uda include yang include number ??

File: database/migrations/2021_09_05_102242_create_verification_tokens_table.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+use Illuminate\Support\Facades\Schema;
+
+class CreateVerificationTokensTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
```
Comment (cakazies) [inline database/migrations/2021_09_05_102242_create_verification_tokens_table.php:14]: naise

File: app/Http/Controllers/CustomerController.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        // if validation success, store the data to database
+        $user = Customers::create([
+            'name' => $request->get('name'),
+            'email' => $request->get('email'),
+            'password' => Hash::make($request->get('password')), //Hash::make is using bcrypt with a work factor of 10
+        ]);
+
+        //get token for this new user based on their credentials
+        $token = JWTAuth::fromUser($user);
```
Comment (cakazies) [inline app/Http/Controllers/CustomerController.php:43]: kalau pake JWTAuth untuk key secretnya pake apa yah ???

File: app/Http/Controllers/CustomerController.php (hunk 1 above)
Comment (codingwill) [reply app/Http/Controllers/CustomerController.php:22]: Yes, udah aku include pake regex ya mas...

File: app/Http/Controllers/CustomerController.php (hunk 3 above)
Comment (codingwill) [reply app/Http/Controllers/CustomerController.php:43]: Untuk key secretnya digenerate dan ditaruh di file .env mas.
Generatornya pakai php artisan, ada function dari jwt untuk generate secret keynya **jwt:secret**.

File: app/Http/Controllers/CustomerController.php (hunk 1 above)
Comment (cakazies) [reply app/Http/Controllers/CustomerController.php:22]: øke sip

File: app/Http/Controllers/CustomerController.php (hunk 3 above)
Comment (cakazies) [reply app/Http/Controllers/CustomerController.php:43]: oke sip2

**Pull request merged**